src/commands/protocols/publish/sender/v1.0.0/v1-0-0-publish-request-command.js

```diff
@@ -6,19 +6,17 @@ class PublishRequestCommand extends ProtocolRequestCommand {
         super(ctx);
         this.operationService = ctx.publishService;
         this.signatureStorageService = ctx.signatureStorageService;
-
+        this.pendingStorageService = ctx.pendingStorageService;
         this.errorType = ERROR_TYPE.PUBLISH.PUBLISH_STORE_REQUEST_ERROR;
     }
 
     async prepareMessage(command) {
-        const { datasetRoot, operationId } = command.data;
+        const { datasetRoot } = command.data;
 
         // TODO: Backwards compatibility, send blockchain without chainId
-        const blockchain = command.data.blockchain.split(':')[0];
+        const { blockchain } = command.data;
 
-        const {
-            public: { dataset },
-        } = await this.operationIdService.getCachedOperationIdData(operationId);
+        const dataset = await this.pendingStorageService.getCachedDataset(blockchain, datasetRoot);
```

Comment:
On publish the node will schedule this command R2 times. This means it will read R2 times from file. This is why memory caching was introduced in operation id service.